src/blister.c

```diff
@@ -1569,14 +1569,110 @@ static Obj FuncMEET_BLIST(Obj self, Obj list1, Obj list2)
     /* compute the difference by operating blockwise                       */
     ptr1 = CONST_BLOCKS_BLIST(list1);
     ptr2 = CONST_BLOCKS_BLIST(list2);
-    for ( i = NUMBER_BLOCKS_BLIST(list1); 0 < i; i-- ) 
-      { 
+    for ( i = NUMBER_BLOCKS_BLIST(list1); 0 < i; i-- )
+      {
         if (*ptr1++ & *ptr2++) return True;
       }
 
     return False;
 }
 
+/****************************************************************************
+**
+*F  FuncFLIP_BLIST( <self>, <list> ) . . .
+**
+**  'FuncFLIP_BLIST' implements the internal function 'FlipBlist'.
+**
+**  'FlipBlist( <list> )'
+**
+**  'FlipBlist' changes every value in the blist <list> from true to false, and
+**  vice versa.
+*/
+
+static Obj FuncFLIP_BLIST(Obj self, Obj list)
+{
+    // get and check the arguments
+    RequireBlist("FlipBlist", list, "blist");
+
+    if (LEN_BLIST(list) == 0) {
+        return 0;
+    }
+
+    UInt * ptr = BLOCKS_BLIST(list);
+    for (UInt i = NUMBER_BLOCKS_BLIST(list); 0 < i; i--) {
+        *ptr = ~(*ptr);
+        ptr++;
+    }
+    // If the logical length of the boolean list is not a multiple of BIPEB the
+    // last block will contain unused bits, which are then zero.
+    UInt mask =
+        (UInt)~0 >> ((BIPEB * NUMBER_BLOCKS_BLIST(list)) - LEN_BLIST(list));
+    ptr = BLOCK_ELM_BLIST_PTR(list, LEN_BLIST(list));
+    *ptr &= mask;
+    return 0;
+}
+
+/****************************************************************************
+**
+*F  FuncCLEAR_ALL_BLIST( <self>, <list> ) . . .
+**
+**  'FuncCLEAR_ALL_BLIST' implements the internal function 'ClearAllBlist'.
+**
+**  'ClearAllBlist( <list> )'
+**
+**  'ClearAllBlist' changes every value in the blist <list> to false.
+*/
+
+static Obj FuncCLEAR_ALL_BLIST(Obj self, Obj list)
+{
+    // get and check the arguments
+    RequireBlist("ClearAllBitsBlist", list, "blist");
+
+    if (LEN_BLIST(list) == 0) {
+        return 0;
+    }
+
+    UInt * ptr = BLOCKS_BLIST(list);
+    for (UInt i = NUMBER_BLOCKS_BLIST(list); 0 < i; i--) {
+        *ptr++ = 0;
+    }
+
+    return 0;
+}
+
+/****************************************************************************
+**
+*F  FuncSET_ALL_BLIST( <self>, <list> ) . . .
+**
+**  'FuncSET_ALL_BLIST' implements the internal function 'SetAllBlist'.
+**
+**  'SetAllBlist( <list> )'
+**
+**  'SetAllBlist' changes every value in the blist <list> to true.
+*/
+
+static Obj FuncSET_ALL_BLIST(Obj self, Obj list)
+{
+    // get and check the arguments
+    RequireBlist("SetAllBitsBlist", list, "blist");
+
+    if (LEN_BLIST(list) == 0) {
+        return 0;
+    }
+
+    UInt * ptr = BLOCKS_BLIST(list);
+    for (UInt i = NUMBER_BLOCKS_BLIST(list); 0 < i; i--) {
+        *ptr++ = (UInt)~0;
```

Comment:
Very strictly and pedantically speaking, `(UInt)~0` (also elsewhere in this PR) should be `~(UInt)0` for full portability, although I assume all machines built today and probably also in the future will be using two's complement, so it probably doesn't matter... (some of the code in this file also uses `~0UL`).

Anyway, I won't complain if this is left as-is, we probably rely on two's complement elsewhere anyway *shrug*

Reply:
Happy to change this also